### ISSUE TITLE
K-medoid method produces single point clusters that contain items within other clusters

### ISSUE THREAD

**Issue by bpostance** [annoviko/pyclustering#593]
I have a minkowski distance matrix and I am using k-medoids to cluster, and then scoring using scikit's silhouette function. 

Scikit requires cluster labels array as input, so I convert pyclustering output using a dataframe. occasionally I get a length error that these labels are greater in length that my input data dimension. 

After digging into it, the clusters returned from k-medoids occasionally contains single point (length = 1) lists and these index points are within other clusters. Increasing the length of cluster labels.

You can see and obtain my data, code and error in this [git commit](https://github.com/bpostance/training.data_science/blob/9ee11e01ac835bcf11fcc136c8197591fbe37766/ML/2.3_Clustering/10-Clustering-Mixed-Data.ipynb)

```
from pyclustering.cluster.kmedoids import kmedoids
from scipy.spatial.distance import pdist
from scipy.spatial.distance import squareform

k_cand = [3,7,15,30,45,60]

for k in k_cand:
    # initiate k random medoids - sets k clusters
    initial_medoids = np.random.randint(0,1000,size=k)
    kmedoids_instance = kmedoids(Xy_scaled_minkowski,initial_medoids, data_type='distance_matrix')    

    # run cluster analysis and obtain results
    %time kmedoids_instance.process()
    clusters = kmedoids_instance.get_clusters()
    medoids = kmedoids_instance.get_medoids()

    # convert cluster output
    cluster_array = pd.DataFrame([(x,e) for e,i in enumerate(clusters) for x in i]).sort_values(by=0)[1].values
    
    # score
    score1 = silhouette_score(Xy_scaled_minkowski, cluster_array, metric='precomputed')
```

### Quick fix by only considering clusters with 
len(k)>1

```
    # convert cluster output
    cluster_array = pd.DataFrame([(x,e) for e,i in enumerate(clusters) for x in i if len(i)>1]).sort_values(by=0)[1].values
```

**Comment by annoviko** [comment]
The issue is corrected since `0.10.1`.

**Comment by annoviko** [comment]
The issue duplicates https://github.com/annoviko/pyclustering/issues/503 .
